x-pack/legacy/plugins/uptime/public/state/api/monitor_status.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// import { ThrowReporter } from 'io-ts/lib/ThrowReporter';
+// import { isRight } from 'fp-ts/lib/Either';
+import { getApiPath } from '../../lib/helper';
+// import { SnapshotType, Snapshot } from '../../../common/runtime_types';
+import { QueryParams } from '../actions/types';
+import { Ping } from '../../../common/graphql/types';
+
+export interface APIParams {
+  basePath: string;
+  monitorId: string;
+}
+
+export const fetchMonitorStatus = async ({
+  basePath,
+  monitorId,
+  dateStart,
+  dateEnd,
+  location,
+}: QueryParams & APIParams): Promise<Ping> => {
+  const url = getApiPath(`/api/uptime/monitor/status`, basePath);
+  const params = {
+    monitorId,
+    dateStart,
+    dateEnd,
+    ...(location && { location }),
+  };
+  const urlParams = new URLSearchParams(params).toString();
```

Comment:
CC @justinkambic I recommended the use of this in another PR, here's an example of its use here. Thought you might be interested.

Reply:
Yep I've implemented the feedback.